crates/core/src/evm/mod.rs

```diff
@@ -0,0 +1,101 @@
+use super::{
+    types::{Account, BlockHeader, Transaction},
+    Address,
+};
+use revm::{
+    inspector_handle_register,
+    inspectors::TracerEip3155,
+    primitives::{BlockEnv, Bytecode, TxEnv, TxKind, U256},
+    CacheState, Evm,
+};
+use std::collections::HashMap;
+// Rename imported types for clarity
+use revm::primitives::AccountInfo as RevmAccountInfo;
+use revm::primitives::Address as RevmAddress;
+// Export needed types
+pub use revm::primitives::ExecutionResult;
+pub use revm::primitives::SpecId;
+
+pub fn execute_tx(
+    tx: &Transaction,
+    header: &BlockHeader,
+    pre: &HashMap<Address, Account>, // TODO: Modify this type when we have a defined State structure
+    spec_id: SpecId,
+) -> ExecutionResult {
+    let block_env = block_env(header);
+    let tx_env = tx_env(tx);
+    let cache_state = cache_state(pre);
+    let mut state = revm::db::State::builder()
+        .with_cached_prestate(cache_state)
+        .with_bundle_update()
+        .build();
+    let mut evm = Evm::builder()
+        .with_db(&mut state)
+        .with_block_env(block_env)
+        .with_tx_env(tx_env)
+        .with_spec_id(spec_id)
+        .reset_handler()
+        .with_external_context(TracerEip3155::new(Box::new(std::io::stderr())).without_summary())
+        .append_handler_register(inspector_handle_register)
+        .build();
+    let tx_result = evm.transact().unwrap();
+    tx_result.result
```

Comment:
perhaps we can map this to our own result type so we don't expose a revm type

Reply:
Looking at the `ExecutionResult` struct from revm, most of the fields are unit enums and basic types so exposing it doesn't seem as hurtful as copying it  over 1 to 1 with all its inner structs. Maybe we can do this when we have a clearer idea of what information we want to include in our execution result as to not do a 1 to 1 copy.

Reply:
+1 to having our own ExecutionResult (even when it is just a copy). 
We can add/replace fields in the future. The benefit of this is that it's cheaper than having a tracking issue to change this in the future. We make it evm-impl agnostic from day 1 and later if we change it to evm-mlir or something else we don't have to take that into account.

Reply:
Done